apps/src/code-studio/assets/SpriteUpload.js

```diff
@@ -58,22 +75,75 @@ export default class SpriteUpload extends React.Component {
     });
   };
 
+  handleAvailabilityChange = event => {
+    this.setState({spriteAvailability: event.target.value});
+  };
+
   render() {
-    const {uploadStatus, filePreviewURL, category} = this.state;
+    const {
+      uploadStatus,
+      filePreviewURL,
+      currentCategories,
+      spriteAvailability,
+      category,
+      filename
+    } = this.state;
+
+    const uploadButtonDisabled =
+      spriteAvailability === '' ||
+      (spriteAvailability === spriteLocation.LIBRARY && category === '') ||
+      filename === '';
+
     return (
       <div>
         <h1>Sprite Upload</h1>
         <form onSubmit={this.handleSubmit}>
           <label>
-            <h4>Sprite Category:</h4>
-            <input
-              type="text"
-              value={category}
-              onChange={this.handleCategoryChange}
-            />
+            <h3>Sprite Category:</h3>
+            <p>
+              Select whether the sprite should only be available in a specific
+              level or whether it should be available in the sprite library.
+            </p>
+            <div>
+              <label>
+                Level-specific sprite:
+                <input
+                  type="radio"
+                  id="levelSprite"
+                  name="spriteAvailability"
+                  style={styles.radioButton}
+                  value={spriteLocation.LEVEL}
+                  onChange={this.handleAvailabilityChange}
+                />
+              </label>
+              <label>
+                Library sprite:
+                <input
+                  type="radio"
+                  id="librarySprite"
+                  name="spriteAvailability"
+                  style={styles.radioButton}
+                  value={spriteLocation.LIBRARY}
+                  onChange={this.handleAvailabilityChange}
+                />
+              </label>
+            </div>
+            {this.state.spriteAvailability === spriteLocation.LIBRARY && (
```

Comment:
`spriteAvailability` was destructured from state earlier, so we can remove `this.state.` here